SimPEG/directives/directives.py

```diff
@@ -2300,24 +2455,47 @@ def update(self):
             else:
                 jtj_diag += sim.getJtJdiag(m, W=dmisfit.W)
 
-        # Normalize and threshold weights
+        # Compute and sum root-mean squared sensitivities for all objective functions
         wr = np.zeros_like(self.invProb.model)
         for reg in self.reg.objfcts:
             if not isinstance(reg, BaseSimilarityMeasure):
                 wr += reg.mapping.deriv(self.invProb.model).T * (
                     (reg.mapping * jtj_diag) / reg.regularization_mesh.vol ** 2.0
                 )
-        if self.normalization:
-            wr /= wr.max()
-        wr += self.threshold
+
         wr **= 0.5
+
+        # Apply thresholding
+        if self.threshold_method == "global":
+            wr += self.threshold_value
+        elif self.threshold_method == "percentile":
+            wr = np.clip(
+                wr, a_min=np.percentile(wr, self.threshold_value), a_max=np.inf
+            )
+        else:
+            wr = np.clip(wr, a_min=self.threshold_value * wr.max(), a_max=np.inf)
+
+        # Apply normalization
+        if self.normalization_method == "maximum":
+            wr /= wr.max()
+        elif self.normalization_method == "minimum":
+            wr /= wr.min()
+
+        # Add sensitivity weighting to all model objective functions
         for reg in self.reg.objfcts:
             if not isinstance(reg, BaseSimilarityMeasure):
                 sub_regs = getattr(reg, "objfcts", [reg])
                 for sub_reg in sub_regs:
                     sub_reg.set_weights(sensitivity=sub_reg.mapping * wr)
 
     def validate(self, directiveList):
+        """Validate directive against directives list.
+
```

Comment:
Should probably make a comment here (and in the class docstring itself) That this class must come before `BetaEstimate_ByEig` and `UpdatePreconditioner` in the directives list.

Reply:
Done